### PR TITLE
[IMP]sale_delivery_state: Skip sale order lines for delivery state

### PR DESCRIPTION
**Pull request by manuelregidor**
Based on https://github.com/OCA/sale-workflow/issues/2818

### CODE REVIEW

**Comment by manuelregidor** [comment]
@HaraldPanten @ValentinVinagre @etobella @yostashiro Could you review, please? Thank you.

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Review by etobella — APPROVED** [review]

**Comment by manuelregidor** [comment]
There is a failing test from sale_by_packaging but it doesn't seem to be related to this imp.

**Comment by HaraldPanten** [comment]
@etobella We made a small change in order to solve the tests issue. Could you re-review, please? 🙏

File: sale_delivery_state/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    skip_sale_delivery_state = fields.Boolean(
+        string="Skip Delivery State",
+        compute="_compute_skip_sale_delivery_state",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("company_id", "product_id")
+    def _compute_skip_sale_delivery_state(self):
+        for sel in self:
```
Comment (etobella) [inline sale_delivery_state/models/sale_order_line.py]: I do usually prefer to use for record in self: or for line in self:

**Review by etobella — APPROVED** [review]
Just a small nitpicking comment,

The main idea of the change is that it is more readable

File: sale_delivery_state/models/sale_order_line.py (hunk 1 above)
Comment (manuelregidor) [reply sale_delivery_state/models/sale_order_line.py]: done

**Review by HaraldPanten — APPROVED** [review]

**Comment by simahawk** [comment]
Reminder for the next time: when adding a field to a model that could be linked to thousands or millions of records, add a pre-mig script to create the field before upgrading the module to avoid issues w/ existing DBs :wink:

**Pull request merged**